enforcer.go

```diff
@@ -754,6 +765,20 @@ func (e *Enforcer) AddNamedDomainMatchingFunc(ptype, name string, fn defaultrole
 	return false
 }
 
+func (e *Enforcer) ensureContextAdapter() (persist.ContextAdapter, error) {
+	contextAdapter, ok := e.adapter.(persist.ContextAdapter)
+	if ok {
+		return contextAdapter, nil
+	}
+
+	adapter, ok := e.adapter.(persist.Adapter)
```

Comment:
If user choose to use context, then the adapter must support context, otherwise you should return an error (ContextAdapter is not supported) instead of continuing return adapter.